libbeat/service/service_windows.go

```diff
@@ -71,10 +81,32 @@ func ProcessWindowsControlEvents(stopCallback func()) {
 	if isInteractive {
 		run = debug.Run
 	}
-	err = run(os.Args[0], &beatService{})
-	if err != nil {
-		logp.Err("Error: %v", err)
-	} else {
+
+	totalSleep := time.Duration(0)
+	for {
+		err = run(os.Args[0], &beatService{})
+		if err != nil {
+			// Sometimes it is the case that this is invoked against a service before that service has finished starting
+			if errnoErr, ok := err.(syscall.Errno); ok && errnoErr == couldNotConnect {
+				logMsg := "Windows control event not sent, perhaps the service is still starting?"
+				time.Sleep(svcConnectIterSleep)
+				totalSleep += svcConnectIterSleep
+
+				if totalSleep < svcConnectMaxSleep {
+					logp.Info("%s Retrying: %v", logMsg, err)
+					continue
+				}
+
+				logp.Info("%s Giving up: %v", logMsg, err)
+			} else {
+				logp.Err("Unexpected error in service_windows.go, please report as a bug: %+v | %T", err, err)
```

Comment:
Seems like this line breaks quite a few other windows tests (which is probably a good thing?)